android/src/main/java/com/swmansion/gesturehandler/core/GestureHandlerOrchestrator.kt

```diff
@@ -60,9 +58,9 @@ class GestureHandlerOrchestrator(
     }
   }
 
-  private inline fun compactHandlersIf(handlers: Array<GestureHandler<*>?>, count: Int, predicate: (handler: GestureHandler<*>?) -> Boolean): Int {
+  private inline fun compactHandlersIf(handlers: ArrayList<GestureHandler<*>?>, predicate: (handler: GestureHandler<*>?) -> Boolean): Int {
```

Comment:
```suggestion
  private inline fun compactHandlersIf(handlers: List<GestureHandler<*>?>, predicate: (handler: GestureHandler<*>?) -> Boolean): Int {
```

There's no point to depend on a specific list type

Reply:
As far as I can see you cannot use index operator `[]` on `List` (and I think that makes sense). Given that, I don't think it is worth to change this methods body just to get rid of `ArrayList` in favor of `List`.